### PR TITLE
[DNM] Limit type-squatting on top gems when creating new rubygems

### PR DESCRIPTION
**Pull request by segiddins**
DNM since I couldn't get tests running locally. Needs new tests added as well, in addition to some performance profiling

### CODE REVIEW

File: app/models/pusher.rb [hunk 1]
```diff
@@ -60,7 +61,14 @@ def find
 
     @rubygem = Rubygem.name_is(name).first || Rubygem.new(name: name)
 
-    unless @rubygem.new_record?
+    if @rubygem.new_record?
+      Rubygem.downloaded(50).each do |rubygem|
+        if Gem::Text.levenstein_distance(name.downcase, rubygem.name.downcase) < 3
```
Comment (indirect) [inline app/models/pusher.rb]: Pretty sure it's `levenshtein_distance`? Unrelated, if it's fast enough I'd like to do the top 100 (or top 1000?). Maybe `Rubygem.downloded(1000).pluck(:name).first{|n| Gem::Text.levenshtein_distance(name.downcase, n.downcase) < 3 }`?

File: app/models/pusher.rb [hunk 2]
```diff
@@ -60,7 +61,14 @@ def find
...
+          return notify("The name #{name.inspect} is too close to #{rubygem.name.inspect}.\n" \
+                        "Please send an email to xxx@rubygems.org if you believe this is an error.", 409)
```
Comment (indirect) [inline app/models/pusher.rb]: I believe `help@rubygems.org` is the Tender support email queue.

**Comment by simi** [comment]
Isn't this "superseeded" by https://github.com/rubygems/rubygems.org/pull/2037 :question:

**Comment by sonalkr132** [comment]
See #2037

**Pull request closed without merging**